libraries-3/src/test/java/com/baeldung/jnats/NatsClientLiveTest.java

```diff
@@ -1,115 +1,336 @@
+    public static final int POLL_DELAY_MILLIS = 50;
+
+    private static Connection createConnection() throws IOException, InterruptedException {
+        return createConnection("nats://localhost:4222");
+    }
 
-        NatsClient client = connectClient();
+    private static Connection createConnection(String uri) throws IOException, InterruptedException {
+        Options options = new Options.Builder().server(uri)
+            .connectionListener((connection, event) -> System.out.println("Connection Event: " + event))
+            .errorListener(new CustomErrorListener())
+            .build();
+        return Nats.connect(options);
+    }
 
-        SyncSubscription fooSubscription = client.subscribeSync("foo.bar");
-        SyncSubscription barSubscription = client.subscribeSync("bar.foo");
-        client.publishMessage("foo.bar", "bar.foo", "hello there");
+    public static Connection createConnectionWithReportNoResponders() throws IOException, InterruptedException {
+        return createConnectionWithReportNoResponders("nats://localhost:4222");
+    }
 
-        Message message = fooSubscription.nextMessage(200);
-        assertNotNull("No message!", message);
-        assertEquals("hello there", new String(message.getData()));
+    public static Connection createConnectionWithReportNoResponders(String uri) throws IOException, InterruptedException {
+        Options options = new Options.Builder().server(uri)
+            .connectionListener((connection, event) -> System.out.println("Connection Event: " + event))
+            .errorListener(new CustomErrorListener())
+            .reportNoResponders()
+            .build();
+        return Nats.connect(options);
+    }
 
-        client.publishMessage(message.getReplyTo(), message.getSubject(), "hello back");
+    static class CustomErrorListener implements ErrorListener {
+        @Override
+        public void errorOccurred(Connection conn, String error) {
+            System.err.println("Error Occurred: " + error);
+        }
 
-        message = barSubscription.nextMessage(200);
-        assertNotNull("No message!", message);
-        assertEquals("hello back", new String(message.getData()));
+        @Override
+        public void exceptionOccurred(Connection conn, Exception exp) {
+            System.err.println("Exception Occurred: " + exp);
+        }
     }
 
+//    public static void main(String[] args) throws Exception {
+//        Connection conn = createConnection("nats://localhost:4222");
+//        Thread.sleep(60000);
+//    }
 
-    private NatsClient connectClient() {
-        return new NatsClient();
+    @Test
+    public void givenNoSubscribers_whenPublishingMessages_thenItCannotBeDeterminedIfMessageIsReceived() {
+        try (Connection natsConnection = createConnection()) {
+            natsConnection.publish("mySubject", "myData".getBytes());
+        }
+        catch (Exception e) {
+            fail("There should not be an exception");
+        }
```

Comment:
We can just let the test fail with the exception itself if you remove the catch clause
```suggestion
```